### PR TITLE
[frontend] fix filter priority order o

### PR DESCRIPTION
**Pull request by savacano28**
### Proposed changes

* Add default values for status, source on paylaod relation
* Fix filter for enumerated types : Add operator: empty, no_empty to enumerate types
Screenshot 2024-09-25 143940
image

### Related issues

* https://github.com/OpenBAS-Platform/openbas/issues/1478

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by RomuDeuxfois** [comment]
Why this changes ?
Now, we can't filter with empty and is empty for several property
image

Additionally, this breaks compatibility with existing filters in local storage that used other operators.
This is a big functional change.

File: openbas-front/src/components/common/queryable/filter/FilterUtils.tsx [hunk 1]
```diff
@@ -103,6 +99,10 @@ export const availableOperators = (propertySchema: PropertySchemaDTO) => {
   if (propertySchema.schema_property_has_dynamic_value && !propertySchema.schema_property_type_array) {
     return ['contains', 'not_contains', 'empty', 'not_empty'];
   }
+  // Array
+  if (propertySchema.schema_property_values || propertySchema.schema_property_type_array) {
+    return ['contains', 'not_contains', 'empty', 'not_empty'];
+  }
```
Comment (RomuDeuxfois) [inline openbas-front/src/components/common/queryable/filter/FilterUtils.tsx:105]: Why this changes ?
Now, we can't filter with empty and is empty for several property
image

Additionally, this breaks compatibility with existing filters in local storage that used other operators.
This is a big functional change.

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: openbas-front/src/components/common/queryable/filter/FilterUtils.tsx (hunk 1 above)
Comment (savacano28) [reply openbas-front/src/components/common/queryable/filter/FilterUtils.tsx:105]: I changed the order of the conditions because this one is less strict than the others: In this cse we have: 
 // Array
  if (propertySchema.schema_property_type_array **|| propertySchema.schema_property_values**) {
    return ['contains', 'not_contains', 'empty', 'not_empty'];
  }
  // Enum & not array
  **if (propertySchema.schema_property_values** && !propertySchema.schema_property_type_array) {
    return ['eq'];
  } .. For the empty case in an enumerated type, I’d prefer to add this option directly within the enumeration.

**Review by damgouj — APPROVED** [review]
Thanks :) !

File: openbas-api/src/test/java/io/openbas/utils/fixtures/PayloadFixture.java [hunk 2]
```diff
@@ -16,8 +16,8 @@ public static Payload createDefaultCommand() {
     command.setContent("cd ..");
     command.setExecutor("PowerShell");
     command.setPlatforms(new Endpoint.PLATFORM_TYPE[]{Endpoint.PLATFORM_TYPE.Windows});
-    command.setSource(MANUAL);
-    command.setStatus(VERIFIED);
+    command.setSource("MANUAL");
```
[the remark this answers is not in the document]
Comment (savacano28) [reply openbas-api/src/test/java/io/openbas/utils/fixtures/PayloadFixture.java:19]: Thanks! I'm going to check!

**Review by damgouj — APPROVED** [review]

**Review by damgouj — APPROVED** [review]

File: openbas-api/src/main/java/io/openbas/rest/payload/form/PayloadCreateInput.java [hunk 3]
```diff
@@ -28,11 +30,11 @@ public class PayloadCreateInput {
 
   @NotBlank(message = MANDATORY_MESSAGE)
   @JsonProperty("payload_source")
-  private String source;
+  private PAYLOAD_SOURCE source;
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/payload/form/PayloadCreateInput.java:33]: I have an internal error when I try to create a payload

**Comment by RomuDeuxfois** [comment]
When I try to filter on Unverified status
image
Nothing happened
image

This is because, on the front-end if the status is empty we display Unverified.
image

I think we need to manage this filter with something like _handleCustomFilter_

**Comment by savacano28** [comment]
@RomuDeuxfois I've added the fixes. Thank you!
